datadog/dogstatsd/base.py

```diff
@@ -204,24 +250,39 @@ def resolve_host(host, use_default_route):
 
         return get_default_route()
 
-    def get_socket(self):
+    def get_socket(self, telemetry=False):
```

Comment:
I think we should split this function to avoid code duplication. We could have `get_socket()` and `get_telemetry_socket()`. And then 2 helper fuctions `get_udp_socket(host, port)` and `get_uds_socket(socket_path)`

Reply:
Coolio.